modules/calib3d/src/calibration.cpp

```diff
@@ -1529,7 +1628,35 @@ CV_IMPL double cvCalibrateCamera2( const CvMat* objectPoints,
         std::copy(param + 4, param + 4 + 14, k);
 
         if( !proceed )
+        {
+            //do errors estimation
+            if(stdDevs)
+            {
+                Ptr<CvMat> JtJ(cvCreateMat(nparams, nparams, CV_64F));
+                CvMat cvMatM(matM);
+                cvEvaluateJtJ2(JtJ, &matA, &_k, &cvMatM, solver.param, npoints, flags, NINTRINSIC, aspectRatio);
```

Comment:
yes, you have to make sure that there is one more update after the solver stops. But for this you just have to move the break down and slightly adapt the ifs.